dashboard/src/actions/repos.ts

```diff
@@ -119,33 +119,45 @@ export const installRepo = (
   name: string,
   repoURL: string,
   authHeader: string,
+  customCA: string,
 ): ThunkAction<Promise<boolean>, IStoreState, null, AppReposAction> => {
   return async (dispatch, getState) => {
     try {
       const {
         config: { namespace },
       } = getState();
-      let auth;
+      const auth: { header?: any; customCA?: any } = {};
+      const secrets: { [s: string]: string } = {};
       const secretName = `apprepo-${name}-secrets`;
-      if (authHeader.length) {
+      if (authHeader.length || customCA.length) {
         // ensure we can create secrets in the kubeapps namespace
-        auth = {
-          header: {
+        if (authHeader.length) {
+          auth.header = {
             secretKeyRef: {
               key: "authorizationHeader",
               name: secretName,
             },
-          },
-        };
+          };
+          secrets.authorizationHeader = btoa(authHeader);
+        }
+        if (customCA.length) {
+          auth.customCA = {
+            secretKeyRef: {
+              key: "ca.crt",
+              name: secretName,
+            },
+          };
+          secrets["ca.crt"] = btoa(customCA);
```

Comment:
We might want to be consistent in the way we reference and name the keys

```
secrets.authorizationHeader vs secrets["ca.crt"]
```

Reply:
the latter is needed because of the `.` in the name of the key, I think it's fine to use the hash notation in that case, but prefer dot notation when we can

Reply:
yes, it's because of the `.`. Note that in other case the linter corrects you so it's not possible to set `secrets["authorizationHeader"]`